llvm/lib/Analysis/StackSafetyAnalysis.cpp

```diff
@@ -233,6 +237,41 @@ SSFunctionSummary::SSFunctionSummary(FunctionSummary &FS) {
   }
 }
 
+SSFunctionSummary::SSFunctionSummary(GlobalAlias &A) {
+  assert(isa<Function>(A.getAliasee()));
+
+  this->GV = &A;
+  this->GVS = nullptr;
+  this->DSOLocal = A.isDSOLocal();
+  this->Interposable = A.isInterposable();
+
+  // 'Forward' all parameters to this alias to the aliasee
+  Function *DestFn = cast<Function>(A.getAliasee());
+
+  for (unsigned ArgNo = 0; ArgNo < DestFn->arg_size(); ArgNo++) {
+    this->Params.emplace_back();
+    SSUseSummary &US = this->Params.back().Summary;
+    US.Calls.emplace_back(DestFn, ArgNo);
+  }
+}
+
+SSFunctionSummary::SSFunctionSummary(AliasSummary &AS) {
+  assert(isa<FunctionSummary>(AS.getAliasee()));
+
+  this->GV = nullptr;
+  this->GVS = &AS;
+  this->DSOLocal = AS.isDSOLocal();
+  // Non-prevailing definitions are not marked live.
+  this->Interposable = !AS.isLive();
+
+  FunctionSummary &DestFn = cast<FunctionSummary>(AS.getAliasee());
```

Comment:
Do you need to stripPointerCasts from aliasee?
Please test alias-of-alias, where both aliasees have bitcasts.

Reply:
I added a testcase with a bitcasted alias and added a stripPointerCasts() to the cast for the Alias::getAliasee() in the other constructor (in this constructor FunctionSummary::getAliasee() always returns a GlobalValueSummary)

Not sure what you mean by alias-of-alias?